homeassistant/components/system_bridge/sensor.py

```diff
@@ -1,40 +1,213 @@
     DEVICE_CLASS_TEMPERATURE,
     DEVICE_CLASS_TIMESTAMP,
     DEVICE_CLASS_VOLTAGE,
     ELECTRIC_POTENTIAL_VOLT,
     FREQUENCY_GIGAHERTZ,
+    FREQUENCY_MEGAHERTZ,
     PERCENTAGE,
+    POWER_WATT,
     TEMP_CELSIUS,
 )
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import StateType
 
 from . import SystemBridgeDeviceEntity
 from .const import DOMAIN
 from .coordinator import SystemBridgeDataUpdateCoordinator
 
-ATTR_AVAILABLE = "available"
-ATTR_FILESYSTEM = "filesystem"
-ATTR_LOAD_AVERAGE = "load_average"
-ATTR_LOAD_IDLE = "load_idle"
-ATTR_LOAD_SYSTEM = "load_system"
-ATTR_LOAD_USER = "load_user"
-ATTR_MOUNT = "mount"
-ATTR_SIZE = "size"
-ATTR_TYPE = "type"
-ATTR_USED = "used"
+ATTR_AVAILABLE: Final = "available"
+ATTR_FILESYSTEM: Final = "filesystem"
+ATTR_MOUNT: Final = "mount"
+ATTR_SIZE: Final = "size"
+ATTR_TYPE: Final = "type"
+ATTR_USED: Final = "used"
+
+
+@dataclass
+class SystemBridgeSensorEntityDescription(SensorEntityDescription):
+    """Class describing System Bridge sensor entities."""
+
+    value: Callable = round
+
+
+BASE_SENSOR_TYPES: tuple[SystemBridgeSensorEntityDescription, ...] = (
+    SystemBridgeSensorEntityDescription(
+        key="bios_version",
+        name="BIOS Version",
+        entity_registry_enabled_default=False,
+        icon="mdi:chip",
+        value=lambda bridge: bridge.system.bios.version,
+    ),
+    SystemBridgeSensorEntityDescription(
+        key="cpu_speed",
+        name="CPU Speed",
+        state_class=STATE_CLASS_MEASUREMENT,
+        native_unit_of_measurement=FREQUENCY_GIGAHERTZ,
+        icon="mdi:speedometer",
+        value=lambda bridge: bridge.cpu.currentSpeed.avg,
+    ),
+    SystemBridgeSensorEntityDescription(
+        key="cpu_temperature",
+        name="CPU Temperature",
+        entity_registry_enabled_default=False,
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value=lambda bridge: bridge.cpu.temperature.main,
+    ),
+    SystemBridgeSensorEntityDescription(
+        key="cpu_voltage",
+        name="CPU Voltage",
+        entity_registry_enabled_default=False,
+        device_class=DEVICE_CLASS_VOLTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        value=lambda bridge: bridge.cpu.cpu.voltage,
+    ),
+    SystemBridgeSensorEntityDescription(
+        key="kernel",
+        name="Kernel",
+        state_class=STATE_CLASS_MEASUREMENT,
+        icon="mdi:devices",
+        value=lambda bridge: bridge.os.kernel,
+    ),
+    SystemBridgeSensorEntityDescription(
+        key="memory_free",
+        name="Memory Free",
+        state_class=STATE_CLASS_MEASUREMENT,
+        native_unit_of_measurement=DATA_GIGABYTES,
+        icon="mdi:memory",
+        value=lambda bridge: round(bridge.memory.free / 1000 ** 3, 2)
+        if bridge.memory.free is not None
```

Comment:
We could consider catching `ValueError` in the `native_value` property instead to clean up the lambdas here.